rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java

```diff
@@ -307,9 +307,9 @@ private boolean isTag(String name) {
     private @Nullable ResolvedManagedDependency findManagedDependency(String groupId, String artifactId, @Nullable String classifier, @Nullable String type) {
         for (ResolvedManagedDependency d : getResolutionResult().getPom().getDependencyManagement()) {
             if (groupId.equals(d.getGroupId()) &&
-                artifactId.equals(d.getArtifactId()) &&
-                (classifier == null || classifier.equals(d.getClassifier())) &&
-                (type == null || type.equals(d.getType()))) {
+                    artifactId.equals(d.getArtifactId()) &&
+                    (classifier == null || classifier.equals(d.getClassifier())) &&
+                    (type == null || type.equals(d.getType()))) {
```

Comment:
```suggestion
                artifactId.equals(d.getArtifactId()) &&
                Objects.equals(classifier, d.getClassifier()) &&
                d.getType().equals(type == null ? "jar" : type)) {
```

Reply:
Not applied here yet; could come in a follow up still, but the added diagnostics here on the exception ought to help already.